src/widgets/modal/EditModal.tsx

```diff
@@ -0,0 +1,83 @@
+import {
+  Button,
+  Image,
+  StyleSheet,
+  TextInput,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import React, {useCallback, useRef} from 'react';
+
+export default function EditModal({
+  addItem,
+  closeModal,
+}: {
+  addItem: (ToDoListItem: any) => void;
+  closeModal: () => void;
+}) {
+  const inputRef = useRef<string>('');
+
+  const handleAddItem = useCallback(
+    function addItemFunc() {
+      const task = inputRef.current;
+      if (task?.trim()) {
+        addItem({text: task});
+        closeModal();
+      }
+    },
+    [addItem, closeModal],
+  );
+  return (
+    <View style={styles.container}>
+      <View style={styles.modalContainer}>
+        <TouchableOpacity onPress={closeModal}>
+          <Image
+            style={styles.xButton}
+            source={require('../../assets/xbutton.png')}
+          />
+        </TouchableOpacity>
+        <TextInput
+          style={styles.inputContainer}
+          onChangeText={text => {
+            inputRef.current = text;
+          }}
```

Comment:
Please use memo to avoid re-creation of the arrow function on every render

Reply:
useCallback